OpenRA.Mods.D2k/Traits/Sandworm.cs

```diff
@@ -111,6 +111,10 @@ void RescanForTargets(Actor self)
 
 			while (!self.World.Map.Contains(moveTo) || !mobile.CanEnterCell(moveTo, null, false))
 			{
+				// without this check, this while statement can be infinity loop
```

Comment:
It would be better to move the `(moveTo == self.Location)` check below inside this loop instead.  That will catch the error case faster, and cancel the activity before returning.